docs/proposals/override-values-inside-configurations-of-different-data-formats/README.md

```diff
@@ -0,0 +1,415 @@
+---
+title: Overriding values inside Configurations of Different Data Formats
+authors:
+- "@Patrick0308"
+- "@sophiefeifeifeiya"
+reviewers:
+- "@chaunceyjiang"
+- TBD
+
+approvers:
+- "@chaunceyjiang"
+- TBD
+
+creation-date: 2024-08-12
+
+
+---
+
+# Overriding values inside Configurations of Different Data Formats
+## Summary
+The proposal introduces a new feature that allows users to partially override configurations of different data formats (e.g., JSON, YAML, XML) by data-format specific operations. This feature expands the current capabilities of `PlaintextOverrider` which only supports whole-field replacements. Take JSON as an example, we can use JSON patch and JSON merge to override the configurations, rather than give a whole JSON file to replace.
+## Motivation
```

Comment:
Can you give an example here to explain why we need this proposal in detail, such as 
- what is the problem this proposal hopes to solve
- why this problem can not be handled by the current `PlaintextOverrider`